Move broadcasting code for fmod, fmod_ from codegen layer.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #37615 Port existing zero_dim_dispatch optimizations from codegen and remove codegen capability.
* #37614 Kill broadcasting from the codegen layer.
* #37613 Move addmm broadcasting code from codegen layer to native layer.
* #37603 Move addbmm broadcasting from the codegen layer to native layer.
* #37582 Specify _th_ ops in CUDAUnaryOps macros so they are easier to find.
* #37547 Kill the ability to codegen tensor-based broadcasting.
* #37546 Move addr broadcasting from codegen layer to native layer.
* **#37545 Move broadcasting code for fmod, fmod_ from codegen layer.**
* #37544 Move baddbmm broadcasting from codegen layer to native layer.

Differential Revision: [D21315036](https://our.internmc.facebook.com/intern/diff/D21315036)

gchanan committed:
Move broadcasting code for fmod, fmod_ from codegen layer.

[ghstack-poisoned]

bhosmer commented on aten/src/ATen/native/cuda/LegacyDefinitions.cpp:
```diff
@@ -69,4 +69,33 @@ Tensor gather_cuda(const Tensor & self, int64_t dim, const Tensor & index, bool
+}
+
+Tensor & fmod_cuda_out(Tensor & result, const Tensor & self, const Tensor & other) {
+  std::tie(b_self, b_other) = expand_outplace(self, other, "fmod_out");
```
missing decls for these I think

gchanan replied:
yep, I didn't copy paste enough :).

gchanan replied:
and I usually don't actually build/test cuda because it takes too long...

gchanan committed:
Update on "Move broadcasting code for fmod, fmod_ from codegen layer."


Differential Revision: [D21315036](https://our.internmc.facebook.com/intern/diff/D21315036)

[ghstack-poisoned]

gchanan committed:
Update on "Move broadcasting code for fmod, fmod_ from codegen layer."


Differential Revision: [D21315036](https://our.internmc.facebook.com/intern/diff/D21315036)

[ghstack-poisoned]

gchanan committed:
Update on "Move broadcasting code for fmod, fmod_ from codegen layer."


Differential Revision: [D21315036](https://our.internmc.facebook.com/intern/diff/D21315036)

[ghstack-poisoned]
